### ISSUE TITLE
Templated class compiles in Jakt, but the C++ stage fails

### ISSUE THREAD

**Issue by mnlrsn** [SerenityOS/jakt#428]
A templated struct is shown in samples/generics/generic_struct.jakt. It compiles and runs. However, if class is substituted for struct (see code below), it fails in the C++ stage. 
```
class A<T> {
    a: T
}

function main() {
    let b = A(a: 0f64)
}
```

**Comment by mnlrsn** [comment]
Slightly updated test case:

```
class A<T> {
    public a: T
}

function main() {
    let b = A(a: 0f64)
    println("{}", b.a)
}
```
Error message:
```
build/jakt -cr ../jakt-mv/Test-class-template-1.jakt
build/Test-class-template-1.cpp:17:24: error: no member named 'a' in 'Jakt::NonnullRefPtr<Jakt::A<double> >'; did you mean to use '->' instead of '.'?
outln(String("{}"),((b).a));
                       ^
                       ->
1 error generated.
```
The suggestion from the error message seems to be correct. When manually changing the `.` to `->` it works.

Interestingly, it works correctly when running it in the REPL (with `--run`)

**Comment by alimpfard** [comment]
> Interestingly, it works correctly when running it in the REPL

The interpreter makes little distinction between classes and structs beyond how they're copied, and it doesn't go through the codegen stage anyway, so it can't hit this bug :)

**Comment by 0GreenClover0** [comment]
Fixed for member variables in #1122 and for methods in #1126
